README.md

```diff
@@ -103,78 +104,155 @@ ibm_cloud_service = QiskitRuntimeService(channel="ibm_cloud", token="MY_IBM_CLOU
+from qiskit.circuit import Parameter
 
 service = QiskitRuntimeService()
 options = Options(optimization_level=1)
 options.execution.shots = 1024  # Options can be set using auto-complete.
 
-psi1 = RealAmplitudes(num_qubits=2, reps=2)
-H1 = SparsePauliOp.from_list([("II", 1), ("IZ", 2), ("XI", 3)])
-theta1 = [0, 1, 1, 2, 3, 5]
+# 1. A quantum circuit for preparing the quantum state (|000> + e^{itheta} |111>)/rt{2}
+theta = Parameter('θ')
+qc_example = QuantumCircuit(3)
+qc_example.h(0) # generate superpostion
+qc_example.p(theta,0) # add quantum phase
+qc_example.cx(0,1) # condition 1st qubit on 0th qubit
+qc_example.cx(0,2) # condition 2nd qubit on 0th qubit
+
+# 2. the observable to be measured
+M1 = SparsePauliOp.from_list([("XXY", 1), ("XYX", 1), ("YXX", 1), ("YYY", -1)])
 
-with Session(service=service, backend="ibmq_qasm_simulator") as session:
-    estimator = Estimator(session=session, options=options)
 
-    # calculate [ <psi1(theta1)|H1|psi1(theta1)> ]
-    job = estimator.run(circuits=[psi1], observables=[H1], parameter_values=[theta1])
-    print(f"Job ID is {job.job_id()}")
-    print(f"Job result is {job.result()}")
+gr = (np.sqrt(5) + 1) / 2 # golden ratio   
+thetaa = 0 # lower range of theta
+thetab = 2*np.pi # upper range of theta
+tol = 1e-1 # tol 
 
-    # You can make additional calls to Sampler and/or Estimator.
+# 3. Execute iteratively using the Estimator primitive 
+with Session(service=service, backend="ibmq_qasm_simulator") as session:
+    estimator = Estimator(session=session, options=options)
+    #next test range 
+    thetac = thetab - (thetab - thetaa) / gr
+    thetad = thetaa + (thetab - thetaa) / gr
+    while abs(thetab - thetaa) > tol:
+        print(f"max value of M1 is in the range theta = {[thetaa, thetab]}")
+        job = estimator.run(circuits=[qc_example]*2, observables=[M1]*2, parameter_values=[[thetac],[thetad]])
+        test =job.result().values
+        if test[0] > test[1]:
+            thetab = thetad
+        else:
+            thetaa = thetac
+        thetac = thetab - (thetab - thetaa) / gr
+        thetad = thetaa + (thetab - thetaa) / gr
+        
+    # Final Job to evaluate estimator at mid point found using golden search method 
+    theta_mid = (thetab + thetaa) / 2
+    job = estimator.run(circuits=qc_example, observables=M1, parameter_values=theta_mid)
+    print(f"Session ID is {session.session_id}")
+    print(f"Final Job ID is {job.job_id()}")
+    print(f"Job result is {job.result().values} at theta = {theta_mid}")
 ```
+This code returns `Job result is [4.] at theta = 1.575674623307102` using only 9 iterations. This is a very powerful extensions to the primitives however beware as using too much code between itertive calls can lock the QPU and use excessive QPU time which is expensive. We recomend only using the session when needed. The sampler can also be used within a session however there are not any well define examples for this. 
```

Comment:
```suggestion
This code returns `Job result is [4.] at theta = 1.575674623307102` using only 9 iterations. This is a very powerful extension to the primitives; however, beware of using too much code between iterative calls, since that can lock the QPU and use excessive QPU time, which is expensive. We recommend only using the session when needed. The Sampler can also be used within a session, but there are not any well-defined examples for this. 
```